[CDAP-13562] Implements profile enabling/disabling in UI

tonybach opened the pull request:
JIRA: https://issues.cask.co/browse/CDAP-13562
BUILD: https://builds.cask.co/browse/CDAP-UDUT1

Things done in this PR:
- Adds a toggle widget in the profile detail view to allow enabling/disabling, but don't show it for 'default' profile, since it cannot be disabled. 
- Shows a confirmation modal when the user wants to disable profile.
- Adds a 'Status' column in the profiles list view and profile preview to show current enabled/disabled status. In the list view, the disabled profiles are greyed out.
- Automatically switches profile selection to 'default' profile, if previously selected profile is now disabled. But for current schedules/triggers, still show the disabled profile, until the user unschedules/deletes the trigger.

tonybach committed:
[CDAP-13562] Implements profile enabling/disabling in UI
- Shows enabled/disabled status in profiles list view table
- Adds status toggle widget in profile detail view using ToggleSwitch component, and modifies it to accept custom on/off labels
- Adds enabled/disabled status column to profiles list view in pipelines, greys out disabled profile, and automatically switches to default when previously selected profile is disabled
- Don't show enable/disable profile switch for the system 'default' profile
- Shows profile status column in profile preview
- Don't change selection to default in the profiles list view is disabled, and sets default profile selection to 'default' instead of showing 'Select a Profile'
- Fixes the styles for angular switch directive to be within the component
- Shows confirmation modal when disabling a profile, and checks profileIsEnabled inside onProfileSelectHandler
- Extracts profile status toggle component

ajainarayanan commented on cdap-ui/app/cdap/components/PipelineDetails/ProfilesListView/index.js:
```diff
@@ -180,30 +223,32 @@ export default class ProfilesListViewInPipeline extends Component {
           */
         }
-        <div onClick={this.onProfileSelectWithoutCustomization.bind(this, profileName)}>
+        <div onClick={profileIsEnabled && onProfileSelectHandler}>
```
Can we do the check inside the `onProfileSelectHandler`. Could make it simpler.

ajainarayanan reviewed (commented):
Just a few minor comments.
I thought we wanted a confirmation modal while disabling/enabling the profile. By design we don't want the confirmation modal?

ajainarayanan commented:
I feel we can make the ToggleSwitch simpler. We can use flexbox and change the order of toggle-button between enabled and disabled state instead of absolute positioning.

ajainarayanan commented:
@tonybach I have made some minor changes for the toggle switch. let me know if the changes are ok.

tonybach commented:
@ajainarayanan Have made minor changes to your changes, so that the styling of the toggle switch component is the same in both Angular and React. Have also added showing confirmation modal when disabling a profile, and moved the `profileIsEnabled` check inside `onProfileSelectHandler`.

ajainarayanan commented:
Talked to @tonybach about moving the profile status toggle to be a separate component.

ajainarayanan reviewed (commented):
👍 Code LGTM

leacd commented:
LGTM

tonybach commented:
On-hold until https://github.com/caskdata/cdap/pull/10233 is merged to avoid creating a merge conflict for that PR.

tonybach commented:
Build https://builds.cask.co/browse/CDAP-UDUT1-8 passed. Merging.
